nl80211: take hostapd ctl iface path from platform_db config file

VladyslavTupikin opened the pull request:
During porting prplMesh to RDKB on RPi it has been founded that
prplMesh assumptions for hostapd ctrl ifaces paths are not
valid for RPi.

Instead of "/var/run/hostapd/<wlan-iface-name>" the paths are
```
    /var/run/hostapd0/wlan0
    /var/run/hostapd1/wlan1
    /var/run/hostapd4/wlan2
```
Ned to rework prplmesh_platform_db file - make it autogenerated by CMake. And modify BWL and agent components for using it, 

Author of original text Ivan Efimov, #1364. Original PR was closed because changed a lot and need to discuss new changes. Discussion in new PR help skips useless commentaries from previous work. 

https://jira.prplfoundation.org/browse/PPM-232

Commit:
bwl: dummy: Modify mon_wlan_hal_dummy constructor

Need to update mon_wlan_hal_create() method
by adding argument for hal_conf_t structure,
and update dummy HAL for using new argument.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>
Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
bwl: dwpal: Modify mon_wlan_hal_dwpal constructor

Need to update mon_wlan_hal_create() method
by adding argument for hal_conf_t structure,
and update dwpal HAL for using new argument.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>
Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
bwl: nl80211: Modify mon_wlan_hal_nl80211 constructor

Need to update mon_wlan_hal_create() method
by adding argument for hal_conf_t structure,
and update nl80211 HAL for using new argument.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>
Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
bwl: dummy: Modify sta_wlan_hal_dummy constructor

Need to update sta_wlan_hal_create() method
by adding argument for hal_conf_t structure,
and update dummy HAL for using new argument.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>
Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
bwl: dwpal: Modify sta_wlan_hal_dwpal constructor

Need to update sta_wlan_hal_create() method
by adding argument for hal_conf_t structure,
and update dwpal HAL for using new argument.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>
Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
bwl: nl80211: Modify sta_wlan_hal_nl80211 constructor

Need to update sta_wlan_hal_create() method
by adding argument for hal_conf_t structure,
and update nl80211 HAL for using new argument.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>
Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
bwl: nl80211: Update initialization for m_wpa_ctrl_path

Need to remove useless condition and change
initialization for m_wpa_ctrl_path variable
from hal_conf_t structure.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
framework: bpl: Create stub for get_ctrl_path methods

Need to create 2 stubs for 2 methods:
bpl_cfg_get_wpa_supplicant_ctrl_path
bpl_cfg_get_hostapd_ctrl_path

Create stubs and prototypes with documentation
for the new methods for linux and uci platforms.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
framework: bpl: Add implementation for bpl_cfg_get_wpa_supplicant_ctrl_path()

Add implementation in linux platform for
bpl_cfg_get_wpa_supplicant_ctrl_path() method.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
framework: bpl: Add implementation for bpl_cfg_get_hostapd_ctrl_path()

Add implementation in lunux platform for
bpl_cfg_get_hostapd_ctrl_path() method.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
framework: bpl: Add generation for prplmesh_platform_db file

Need to create new folder with cmake and prplmesh_platform_db.in
files, modify prplmesh_platform_db.in by adding new variables for
hostapd control path and wpa supplicant control path.

Add 6 variables 3 for hostapd and 3 for wpa supplicant, modify
cmake file for building bpl by adding directory with new cmake
file for generation prplmesh_platform_db file.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: scripts: Remove creating prplmesh_platform_db

Need to remove prplmesh_platform_db creation
from the prplmesh_platform_db_init() function
because now prplmesh_platform_db generates by
cmake.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: fronthaul_manager: Add wpa_ctrl_path in hal_conf

Need to add correct wpa_ctrl_path for ap_manager
(for AP manager it's hostapd_ctrl_path) using method
from the bpl library.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: bwl: Add wpa_ctrl_path as a part of hal_conf structure

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
bwl: dummy: Modify ap_wlan_hal_dummy constructor

Need to update ap_wlan_hal_dummy() method
by adding argument for hal_conf_t structure,
and update dummy HAL for using new argument.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
bwl: dwpal: Modify ap_wlan_hal_dwpal constructor

Need to update ap_wlan_hal_dwpal() method
by adding argument for hal_conf_t structure,
and update dummy HAL for using new argument.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
bwl: nl80211: Modify ap_wlan_hal_nl80211 constructor

Need to update ap_wlan_hal_nl80211() method
by adding argument for hal_conf_t structure,
and update dummy HAL for using new argument.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: fronthaul_manager: Modify monitor_thread

Add wpa_ctrl_path object of hal_conf structure and
fill up using bpl method bpl_cfg_get_hostapd_ctrl_path.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: backhaul_manager: Update WPA_ATTACH handler

Need to update WPA_ATTACH handler - create ojcect
of hal_conf_t structure, fill up wpa_ctrl_path using
bpl_cfg_get_wpa_supplicant_ctrl_path() method,
tranfer hal_conf_t object to the sta_wlan_hal_create() method.

Also need to update arguments in sta_wlan_hal_create()
method - add new argument for hal_conf_t structure.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

MuteSpirit commented on common/beerocks/bwl/nl80211/sta_wlan_hal_nl80211.cpp:
```diff
@@ -16,8 +16,10 @@
-    : base_wlan_hal(), base_wlan_hal_nl80211(bwl::HALType::Station, iface_name, callback, {})
+sta_wlan_hal_nl80211::sta_wlan_hal_nl80211(const std::string &iface_name, hal_event_cb_t callback,
+                                           const bwl::hal_conf_t &hal_conf)
+    : base_wlan_hal(),
```
`base_wlan_hal ctor` also accept `hal_conf` as one of arguments:
```
    base_wlan_hal(HALType type, std::string iface_name, IfaceType iface_type,
                  hal_event_cb_t callback, hal_conf_t hal_conf = {});
```
So pass `hal_conf`into `base_wlan_hal` constructor too

VladyslavTupikin replied:
Fixed

MuteSpirit commented on common/beerocks/bwl/nl80211/mon_wlan_hal_nl80211.cpp:
```diff
@@ -88,9 +88,10 @@ static void calc_curr_traffic(std::string buff, uint64_t &total, uint32_t &curr)
-mon_wlan_hal_nl80211::mon_wlan_hal_nl80211(const std::string &iface_name, hal_event_cb_t callback)
+mon_wlan_hal_nl80211::mon_wlan_hal_nl80211(const std::string &iface_name, hal_event_cb_t callback,
+                                           const bwl::hal_conf_t &hal_conf)
     : base_wlan_hal(bwl::HALType::Monitor, iface_name, IfaceType::Intel, callback),
```
`base_wlan_hal ctor` also accept `hal_conf` as one of arguments:
```
    base_wlan_hal(HALType type, std::string iface_name, IfaceType iface_type,
                  hal_event_cb_t callback, hal_conf_t hal_conf = {});
```
So pass `hal_conf`into `base_wlan_hal` constructor too

VladyslavTupikin replied:
Fixed

MuteSpirit commented on common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp:
```diff
@@ -113,9 +113,9 @@ static uint8_t wpa_bw_to_beerocks_bw(const std::string &chan_width)
-    : base_wlan_hal(bwl::HALType::AccessPoint, iface_name, IfaceType::Intel, callback, hal_conf),
-      base_wlan_hal_nl80211(bwl::HALType::AccessPoint, iface_name, callback, BUFFER_SIZE)
+                                         const hal_conf_t &hal_conf)
+    : base_wlan_hal(bwl::HALType::AccessPoint, iface_name, IfaceType::Intel, callback),
```
`base_wlan_hal ctor` also accept `hal_conf` as one of arguments:
```
    base_wlan_hal(HALType type, std::string iface_name, IfaceType iface_type,
                  hal_event_cb_t callback, hal_conf_t hal_conf = {});
```
So pass `hal_conf`into `base_wlan_hal` constructor too

VladyslavTupikin replied:
Fixed

MuteSpirit commented on common/beerocks/bwl/dwpal/sta_wlan_hal_dwpal.cpp:
```diff
@@ -91,9 +91,10 @@ static std::string dwpal_security_val(WiFiSec sec)
-sta_wlan_hal_dwpal::sta_wlan_hal_dwpal(const std::string &iface_name, hal_event_cb_t callback)
+sta_wlan_hal_dwpal::sta_wlan_hal_dwpal(const std::string &iface_name, hal_event_cb_t callback,
+                                       const bwl::hal_conf_t &hal_conf)
     : base_wlan_hal(bwl::HALType::Station, iface_name, IfaceType::Intel, callback),
```
`base_wlan_hal ctor` also accept `hal_conf` as one of arguments:
```
    base_wlan_hal(HALType type, std::string iface_name, IfaceType iface_type,
                  hal_event_cb_t callback, hal_conf_t hal_conf = {});
```
So pass `hal_conf`into `base_wlan_hal` constructor too

VladyslavTupikin replied:
Fixed

MuteSpirit commented on common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp:
```diff
@@ -541,9 +541,10 @@ static std::shared_ptr<char> generate_client_assoc_event(const std::string &even
-mon_wlan_hal_dwpal::mon_wlan_hal_dwpal(const std::string &iface_name, hal_event_cb_t callback)
+mon_wlan_hal_dwpal::mon_wlan_hal_dwpal(const std::string &iface_name, hal_event_cb_t callback,
+                                       const bwl::hal_conf_t &hal_conf)
     : base_wlan_hal(bwl::HALType::Monitor, iface_name, IfaceType::Intel, callback),
```
`base_wlan_hal ctor` also accept `hal_conf` as one of arguments:
```
    base_wlan_hal(HALType type, std::string iface_name, IfaceType iface_type,
                  hal_event_cb_t callback, hal_conf_t hal_conf = {});
```
So pass `hal_conf`into `base_wlan_hal` constructor too

VladyslavTupikin replied:
Fixed

MuteSpirit commented on common/beerocks/bwl/dummy/sta_wlan_hal_dummy.cpp:
```diff
@@ -16,9 +16,10 @@
-sta_wlan_hal_dummy::sta_wlan_hal_dummy(const std::string &iface_name, hal_event_cb_t callback)
+sta_wlan_hal_dummy::sta_wlan_hal_dummy(const std::string &iface_name, hal_event_cb_t callback,
+                                       const bwl::hal_conf_t &hal_conf)
     : base_wlan_hal(bwl::HALType::Station, iface_name, IfaceType::Intel, callback, {}),
```
`base_wlan_hal ctor` also accept `hal_conf` as one of arguments:
```
    base_wlan_hal(HALType type, std::string iface_name, IfaceType iface_type,
                  hal_event_cb_t callback, hal_conf_t hal_conf = {});
```
So pass actual `hal_conf`into `base_wlan_hal` constructor instead of default `{}`

VladyslavTupikin replied:
Fixed

MuteSpirit commented on common/beerocks/bwl/dummy/mon_wlan_hal_dummy.cpp:
```diff
@@ -66,9 +66,10 @@ static mon_wlan_hal_dummy::Data dummy_to_bwl_data(const std::string &opcode)
-mon_wlan_hal_dummy::mon_wlan_hal_dummy(const std::string &iface_name, hal_event_cb_t callback)
+mon_wlan_hal_dummy::mon_wlan_hal_dummy(const std::string &iface_name, hal_event_cb_t callback,
+                                       const bwl::hal_conf_t &hal_conf)
     : base_wlan_hal(bwl::HALType::Monitor, iface_name, IfaceType::Intel, callback),
```
`base_wlan_hal ctor` also accept `hal_conf` as one of arguments:
```
    base_wlan_hal(HALType type, std::string iface_name, IfaceType iface_type,
                  hal_event_cb_t callback, hal_conf_t hal_conf = {});
```
So pass `hal_conf`into `base_wlan_hal` constructor too

VladyslavTupikin replied:
Fixed

MuteSpirit commented on framework/platform/bpl/linux/bpl_cfg.cpp:
```diff
@@ -391,5 +391,45 @@ bool cfg_get_unfriendly_device_max_timelife_delay_days(
+    }
+
+    std::copy(local_wpa_ctrl_path.begin(), local_wpa_ctrl_path.end(),
+              std::back_inserter(wpa_ctrl_path));
```
Why not follow next way:
```
wpa_ctrl_path = local_wpa_ctrl_path;
```
It will also create string copy:
https://en.cppreference.com/w/cpp/string/basic_string/operator%3D
```
basic_string& operator=( const basic_string& str );   (1)
....
1) Replaces the contents with a copy of str. If *this and str are the same object, this function has no effect.
```

VladyslavTupikin replied:
fixed

MuteSpirit commented on framework/platform/bpl/linux/bpl_cfg.cpp:
```diff
@@ -391,5 +391,45 @@ bool cfg_get_unfriendly_device_max_timelife_delay_days(
+    param += '=';
+
+    if (cfg_get_param(param, local_hostapd_ctrl_path) < 0) {
+        MAPF_ERR("bpl_cfg_get_hostapd_ctrl_path: Failed to read: " << param);
```
Maybe use special predefined variable `__func__` instead of manual function name duplication?
Something like:
```
__func__ + ": Failed to read: " << param
```

VladyslavTupikin replied:
Fixed

MuteSpirit commented on framework/platform/bpl/linux/bpl_cfg.cpp:
```diff
@@ -391,5 +391,45 @@ bool cfg_get_unfriendly_device_max_timelife_delay_days(
+    param += '=';
+
+    if (cfg_get_param(param, local_wpa_ctrl_path) < 0) {
+        MAPF_ERR("bpl_cfg_get_wpa_supplicant_ctrl_path: Failed to read: " << param);
```
Maybe use special predefined variable `__func__` instead of manual function name duplication?
Something like:
```
__func__ + ": Failed to read: " << param
```

VladyslavTupikin replied:
Fixed

mariomaz replied:
I think MAPF_ERR is defined as LOG(ERROR) and log message already includes file name and line number. That should be enough to locate failing code so there's no need for an extra string literal with function name nor for `__func__`

MuteSpirit commented on framework/platform/bpl/linux/bpl_cfg.cpp:
```diff
@@ -391,5 +391,45 @@ bool cfg_get_unfriendly_device_max_timelife_delay_days(
+    }
+
+    std::copy(local_hostapd_ctrl_path.begin(), local_hostapd_ctrl_path.end(),
+              std::back_inserter(hostapd_ctrl_path));
```
Why not follow next way:
```
hostapd_ctrl_path = local_hostapd_ctrl_path;
```
It will also create string copy:
https://en.cppreference.com/w/cpp/string/basic_string/operator%3D
```
basic_string& operator=( const basic_string& str );   (1)
....
1) Replaces the contents with a copy of str. If *this and str are the same object, this function has no effect.
```

VladyslavTupikin replied:
fixed

MuteSpirit commented on framework/platform/bpl/uci/cfg/bpl_cfg.cpp:
```diff
@@ -459,5 +459,19 @@ bool cfg_get_unfriendly_device_max_timelife_delay_days(
+    std::string path  = "/var/run/hostapd/";
+    hostapd_ctrl_path = path + iface;
+    return true;
+}
```
Does i correctly understand that for case of using UCI instead of `prplmesh_platform_db` wpa/hostapd ctrl path will be still hard-coded?
Maybe add separate task for real get wpa/hostapd ctrl path from UCI variables which are added by prplmesh init script or OpenWrt package?

VladyslavTupikin replied:
As I understood, this is a common path for OpenWRT and we can keep it as is.

MuteSpirit commented on common/beerocks/scripts/prplmesh_utils.sh.in:
```diff
@@ -90,16 +90,6 @@ platform_deinit() {
-        # In certification, agent must retry onboarding indefinitely
-        echo "certification_mode=1"
-        echo "stop_on_failure_attempts=0"
-    } > @TMP_PATH@/prplmesh_platform_db
```
Maybe just copy file to `@TMP_PATH` ?
```
mkdir -p @TMP_PATH@
cp @CMAKE_MULTIAP_OUTPUT_DIRECTORY@/share/prplmesh_platform_db @TMP_PATH@/prplmesh_platform_db
```

VladyslavTupikin replied:
Agree, but correct string should be:

`cp @INSTALL_PATH@/share/prplmesh_platform_db @TMP_PATH@/prplmesh_platform_db`

arnout commented on agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp:
```diff
@@ -1346,9 +1346,18 @@ bool backhaul_manager::backhaul_fsm_wireless(bool &skip_select)
+
+                bwl::hal_conf_t hal_conf;
+
+                if (!beerocks::bpl::bpl_cfg_get_hostapd_ctrl_path(iface, hal_conf.wpa_ctrl_path)) {
```
This should be the wpa_supplicant path, not hostapd path.

VladyslavTupikin replied:
fixed

arnout commented on framework/platform/bpl/CMakeLists.txt:
```diff
@@ -8,6 +8,23 @@ set(MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
 file(GLOB_RECURSE bpl_common_sources ${MODULE_PATH}/common/*.c*)
 
+# Common prplmesh_db values
+set(BEEROCKS_MANAGEMENT_MODE  "Multi-AP-Controller-and-Agent" CACHE STRING "Set management mode")
+set(BEEROCKS_OPERATING_MODE  "Gateway" CACHE STRING "Set operating mode")
+set(BEEROCKS_STOP_ON_FAILURE_ATTEMPTS  0 CACHE INT "In common config stop on failure disabled")
+set(BEEROCKS_CERTIFICATION_MODE  1 CACHE INT "In certification, agent must retry onboarding indefinitely")
```
These ones aren't used anywhere (and I don't think they should be).

VladyslavTupikin replied:
It is used in prplmesh_platform_db.in file

arnout commented on framework/platform/bpl/CMakeLists.txt:
```diff
@@ -8,6 +8,23 @@ set(MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
+set(BEEROCKS_WPA_SUPPLICANT_WLAN3_CTRL_IFACE "/var/run/wpa_supplicant/${BEEROCKS_WLAN3_IFACE}" CACHE PATH "wpa_supplicant ctrl iface path for 3rd WLAN")
+
+# Add prplmesh_platform_db config generation
+add_subdirectory(platform_db)
```
This should only be done for the Linux implementation, not for UCI.

VladyslavTupikin replied:
I disagree with you @arnout! We need a prplmesh_platform_db file in all platforms. In the current master, it generates by prplmesh_utils.sh and it's weird. I think we should generate prplmesh_platform_db  by CMake for all platforms.

arnout commented on framework/platform/bpl/platform_db/prplmesh_platform_db.in:
```diff
@@ -0,0 +1,10 @@
+management_mode=@BEEROCKS_MANAGEMENT_MODE@
+operating_mode=@BEEROCKS_OPERATING_MODE@
+stop_on_failure_attempts=@BEEROCKS_STOP_ON_FAILURE_ATTEMPTS@
+certification_mode=@BEEROCKS_CERTIFICATION_MODE@
```
Ah, here they are used... Well, I don't think they should be. Management mode and operating mode anyway need to be set by the prplmesh_utils script. The other two you can just hardcode here.

Anyway,nitpick.

VladyslavTupikin replied:
I prefer to keep data in CMake variables - if config file autogenerated - it should be fully generated.

mariomaz commented on framework/platform/bpl/include/bpl/bpl_cfg.h:
```diff
@@ -537,6 +538,24 @@ bool cfg_get_max_timelife_delay_days(int &max_timelife_delay_days);
     int &unfriendly_device_max_timelife_delay_days);
 
+/**
+ * @brief Returns the correct WPA Control Path for the given interface.
```
```suggestion
 * @brief Returns configured WPA Control Path for the given interface.
```

VladyslavTupikin replied:
fixed

mariomaz commented on framework/platform/bpl/include/bpl/bpl_cfg.h:
```diff
@@ -537,6 +538,24 @@ bool cfg_get_max_timelife_delay_days(int &max_timelife_delay_days);
+bool bpl_cfg_get_wpa_supplicant_ctrl_path(const std::string &iface, std::string &wpa_ctrl_path);
+
+/**
+ * @brief Returns the correct Hostapd Control Path for the given interface.
```
```suggestion
 * @brief Returns configured Hostapd Control Path for the given interface.
```

VladyslavTupikin replied:
fixed

Commit:
AUTHORS.md: Add new author

Add Ivan Efimov in AUTHORS.md because
half of commits was prepared by Ivan
Efimov.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

mariomaz commented on framework/platform/bpl/uci/cfg/bpl_cfg.cpp:
```diff
@@ -461,6 +461,8 @@ bool cfg_get_unfriendly_device_max_timelife_delay_days(
 
 bool bpl_cfg_get_wpa_supplicant_ctrl_path(const std::string &iface, std::string &wpa_ctrl_path)
 {
+    std::string path = "/var/run/wpa_supplicant/";
```
```suggestion
    const char* path { "/var/run/wpa_supplicant/" };
```
Only assign string literals to const variables
Don't use std::string for string constant expressions
Favor direct brace initialization whenever possible.

VladyslavTupikin replied:
fixed

mariomaz commented on framework/platform/bpl/uci/cfg/bpl_cfg.cpp:
```diff
@@ -468,6 +468,8 @@ bool bpl_cfg_get_wpa_supplicant_ctrl_path(const std::string &iface, std::string
 
 bool bpl_cfg_get_hostapd_ctrl_path(const std::string &iface, std::string &hostapd_ctrl_path)
 {
+    std::string path  = "/var/run/hostapd/";
```
```suggestion
    const char* path { "/var/run/hostapd/" };
```
Only assign string literals to const variables
Don't use std::string for string constant expressions
Favor direct brace initialization whenever possible.

VladyslavTupikin replied:
fixed

mariomaz commented on agent/src/beerocks/fronthaul_manager/ap_manager/ap_manager_thread.cpp:
```diff
@@ -114,6 +115,11 @@ bool ap_manager_thread::create_ap_wlan_hal()
     hal_conf.ap_acs_enabled = acs_enabled;
 
+    if (!beerocks::bpl::bpl_cfg_get_hostapd_ctrl_path(m_iface, hal_conf.wpa_ctrl_path)) {
+        LOG(ERROR) << "Couldn't get hostapd control path";
```
```suggestion
        LOG(ERROR) << "Couldn't get hostapd control path for interface " << m_iface;
```

VladyslavTupikin replied:
fixed

mariomaz commented on agent/src/beerocks/fronthaul_manager/monitor/monitor_thread.cpp:
```diff
@@ -64,9 +65,15 @@ monitor_thread::monitor_thread(const std::string &slave_uds_, const std::string
+    bwl::hal_conf_t hal_conf;
+
+    if (!beerocks::bpl::bpl_cfg_get_hostapd_ctrl_path(monitor_iface, hal_conf.wpa_ctrl_path)) {
+        LOG(ERROR) << "Couldn't get hostapd control path";
```
```suggestion
        LOG(ERROR) << "Couldn't get hostapd control path for interface " << monitor_iface;
```

VladyslavTupikin replied:
fixed

mariomaz commented on common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp:
```diff
@@ -1433,9 +1434,10 @@ bool mon_wlan_hal_dwpal::process_dwpal_nl_event(struct nl_msg *msg)
 } // namespace dwpal
 
 std::shared_ptr<mon_wlan_hal> mon_wlan_hal_create(std::string iface_name,
```
```suggestion
std::shared_ptr<mon_wlan_hal> mon_wlan_hal_create(const std::string &iface_name,
```

VladyslavTupikin replied:
fixed

mariomaz commented on common/beerocks/bwl/include/bwl/mon_wlan_hal.h:
```diff
@@ -65,7 +65,8 @@ class mon_wlan_hal : public virtual base_wlan_hal {
 
 // mon HAL factory types
 std::shared_ptr<mon_wlan_hal> mon_wlan_hal_create(std::string iface_name,
```
```suggestion
std::shared_ptr<mon_wlan_hal> mon_wlan_hal_create(const std::string &iface_name,
```
While you're at it

Changes in this header file shall be done in the same commit as in the different bwl source code files

VladyslavTupikin replied:
fixed

mariomaz reviewed (approved):
Only nitpicks so approving
